connection_manager.go

```diff
@@ -206,6 +256,10 @@ func (cm *connectionManager) createConnection() (*connection, error) {
 }
 
 func (cm *connectionManager) get() (*connection, error) {
+	if cm.connectionCounter.isLessThan(cm.minConnections) {
+		return cm.create()
+	}
```

Comment:
@pguelpa - how about this for a fix to #49? If the node needs a new connection and the current count is less than the minimum, just go ahead and create a new one.